lib/vector-config/src/stdlib.rs

```diff
@@ -115,12 +115,41 @@ where
     }
 }
 
+impl<V> Configurable for BTreeMap<String, V>
+where
+    V: Configurable + Serialize,
+{
+    fn is_optional() -> bool {
+        // A map with required fields would be... an object.  So if you want that, make a struct
+        // instead, not a hashmap.
+        true
+    }
+
+    fn generate_schema(gen: &mut SchemaGenerator, overrides: Metadata<Self>) -> SchemaObject {
+        // We explicitly do not pass anything from the override metadata, because there's nothing to
+        // reasonably pass: if `V` is referencable, using the description for `BTreeMap<String, V>`
```

Comment:
```suggestion
        // reasonably pass: if `V` is referenceable, using the description for `BTreeMap<String, V>`
```

Chrome still says it's misspelled but 🤷

Reply:
Yeah, it's not a word in the English dictionary but it's common enough in software that I decided to use it. I'm gonna change all instances of `referencable` to `referenceable` for clarity, though.

Reply:
It's definitely what gets returned by googling so seems good enough 😄

Reply:
Fixed in 18c22d4b8.